### PR TITLE
Accessibility Tests for Newsletter

### PR DESCRIPTION
**Pull request by ajitava-git**
This PR is for accessibility tests for newsletter.

I have added a new method in the page factory to return all the pages. The pages are being iterated and injected axe-core and run for checking accessibility.

Once the accessibility checks are run, a snapshot is created for every page.

So, next time a new page is introduced, we just need to add it to the page factory, and the accessibility script will do the job.

An accessibility util is added which has the methods to inject, run, and write axe into the tests.

Steps to run the test:
1. Go to the accessibility _tests folder
2. Run the test using python -m pytest test_accessibility.py --snapshot-update (We use --snapshot-update only for the first time to create the snapshot directory)
3. Once the test is run, you can see the accessibility for each page inside the snapshot folder.

output

### CODE REVIEW

File: tests/integration/page_objects/PageFactory.py [hunk 1]
```diff
@@ -29,16 +28,20 @@ def get_page_object(page_name,base_url=conf.base_url_conf.base_url):
...
         elif page_name == "manage articles page":
             test_obj = Managearticles_Page(base_url=base_url)  
         elif page_name == "edit articles page":
-            test_obj = Editarticles_Page(base_url=base_url)  
+            test_obj = Editarticles_Page(base_url=base_url)
         elif page_name == "create newsletter page":
             test_obj = Createnewsletter_Page(base_url=base_url)
         elif page_name == "preview newsletter page":
-            test_obj = Previewnewsletter_Page(base_url=base_url)    
+            test_obj = Previewnewsletter_Page(base_url=base_url)
```
Comment (shivahari) [inline tests/integration/page_objects/PageFactory.py:39]: Can you add a comment here at the end saying any new page added needs to be updated in the `get_all_page_names` method too.

**Review by shivahari — CHANGES_REQUESTED** [review]

File: tests/integration/tests/accessibility_tests/test_accessibility.py
[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/tests/accessibility_tests/test_accessibility.py:1]: I think this was like how we have for ui_tests and api_tests (the directory structure?) under tests directory and another reason was that accessibility tests would have a snapshot folder for snapshot testing. Didn't quite understand "to help pytest find tests". This test is discoverable by pytest.

File: tests/integration/utils/axe.min.js
[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/utils/axe.min.js:1]: Yes @shivahari.. Make sense. Moved it.

File: tests/integration/page_objects/PageFactory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/page_objects/PageFactory.py]: Done.

File: tests/integration/page_objects/Base_Page.py [hunk 2]
```diff
@@ -37,7 +38,7 @@ def is_first_time(self):
 
         return result_flag
 
-class Base_Page(Borg,unittest.TestCase):
+class Base_Page(Borg,unittest.TestCase, Accessibility_Util):
```
[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/page_objects/Base_Page.py]: @shivahari Have implemented the suggested design pattern. Please review.

**Review by shivahari — APPROVED** [review]

**Review by shivahari — CHANGES_REQUESTED** [review]
Can you add `axe_selenium_python`  module to the `requirements.txt` file

**Review by shivahari — CHANGES_REQUESTED** [review]
Can you also add the command to run the test in the `deploy` GitHub workflow here https://github.com/qxf2/newsletter_automation/blob/main/.github/workflows/deploy.yaml#L47

**Comment by ajitava-git** [comment]
> Can you add `axe_selenium_python` module to the `requirements.txt` file

Done @shivahari

**Comment by ajitava-git** [comment]
> Can you also add the command to run the test in the `deploy` GitHub workflow here https://github.com/qxf2/newsletter_automation/blob/main/.github/workflows/deploy.yaml#L47

Done @shivahari and pushed.

**Review by shivahari — APPROVED** [review]

**Comment by rohandudam** [comment]
@ajitava-git Can you please check CI run issue and make sure it will run fine as you touched the github action file with this PR.

**Comment by ajitava-git** [comment]
> @ajitava-git Can you please check CI run issue and make sure it will run fine as you touched the github action file with this PR. Also think about the order of tests run, I feel it better to run the faster tests first before running slower tests like UI tests.

Done @rohandudam! For now, I have kept the order in this way. API Tests followed by Accessibility Tests followed by UI Tests.

**Review by shivahari — CHANGES_REQUESTED** [review]
The test seems to be missing from `tests/accessibility_tests` Ajitava.

**Comment by ajitava-git** [comment]
> @ajitava-git Can you please check CI run issue and make sure it will run fine as you touched the github action file with this PR. Also think about the order of tests run, I feel it better to run the faster tests first before running slower tests like UI tests.

@rohandudam The accessibility tests run fine in CI now.

**Review by rohandudam — APPROVED** [review]

File: tests/integration/tests/accessibility_tests/snapshots/test_accessibility/test_accessibility/chrome/snapshot_output_add articles page.txt
[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/tests/accessibility_tests/snapshots/test_accessibility/test_accessibility/chrome/snapshot_output_add articles page.txt:1]: Done

[the remark this answers is not in the document]
Comment (ajitava-git) [reply tests/integration/tests/accessibility_tests/snapshots/test_accessibility/test_accessibility/chrome/snapshot_output_add articles page.txt:1]: Done

**Review by shivahari — APPROVED** [review]

**Pull request merged**